AnimationsLab/app/src/main/java/gameshop/serkanbal/com/animationslab/MainActivity.java

```diff
@@ -0,0 +1,64 @@
+package gameshop.serkanbal.com.animationslab;
+
+import android.animation.LayoutTransition;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+public class MainActivity extends AppCompatActivity {
+    EditText mEditText;
+    Button mButton, mButton2;
+    TextView mTextView;
+    ViewGroup mViewGroup;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mEditText = (EditText) findViewById(R.id.edittext);
+        mButton = (Button) findViewById(R.id.addbutton);
+        mButton2 = (Button) findViewById(R.id.delete);
+        mTextView = (TextView) findViewById(R.id.textview);
+
+        LayoutTransition l = new LayoutTransition();
+        l.enableTransitionType(LayoutTransition.CHANGING);
```

Comment:
Since you put `android:animateLayoutChanges="true"` in the XML for the LinearLayout, you actually don't need to also set up the layout transition animation here. That one line in XML accomplishes the same thing.